src/client/driver/command-executors/automation.js

```diff
@@ -0,0 +1,119 @@
+import hammerhead from '../deps/hammerhead';
+import testCafeCore from '../deps/testcafe-core';
+import testCafeRunner from '../deps/testcafe-runner';
+import testCafeUI from '../deps/testcafe-ui';
+import ERROR_TYPE from '../../../legacy/test-run-error/type';
+import waitFor from '../utils/wait-for';
+
+var Promise         = hammerhead.Promise;
+var EventEmitter    = testCafeCore.serviceUtils.EventEmitter;
+var xhrBarrier      = testCafeCore.xhrBarrier;
+var positionUtils   = testCafeCore.positionUtils;
+var domUtils        = testCafeCore.domUtils;
+var ClickAutomation = testCafeRunner.get('./automation/playback/click');
+var ProgressPanel   = testCafeUI.ProgressPanel;
+
+
+const PROGRESS_PANEL_TEXT   = 'Waiting for the target element of the next action to appear';
+const CHECK_ELEMENT_DELAY   = 200;
+const CHECK_ELEMENT_TIMEOUT = 10000;
+
+
+function findElement (selector) {
+    return hammerhead.nativeMethods.querySelector.call(document, selector);
+}
+
+function ensureElementExists (selector) {
```

Comment:
I think we should make `selector` always a function. It will be easier to maintain. For string selectors we will just wrap them in the function server-side.